### ISSUE TITLE
The loss function

### ISSUE THREAD

**Issue by hellojialee** [anatolix/keras_Realtime_Multi-Person_Pose_Estimation#19]
In this repo, the loss function is defined as：
`    def eucl_loss(x, y):
        l = K.sum(K.square(x - y)) / batch_size / 2
        return l`
Why dose the loss divide batch_size / 2 not batch_size？ Thanks for any help！

**Comment by hellojialee** [comment]
Btw, I think the code in https://github.com/anatolix/keras_Realtime_Multi-Person_Pose_Estimation/blob/new-generation/training/ds_generators.py#L94 
`batches_x = np.empty(self.image_shape)
                batches_x1 = np.empty(self.mask1_shape)
                batches_x2 = np.empty(self.mask2_shape)
                batches_y1 = np.empty(self.ypafs1_shape)
                batches_y2 = np.empty(self.yheat2_shape)`
is not necessary even if multi-process. The data will be refreshed in the same memory location every batch.

**Comment by anatolix** [comment]
> Why dose the loss divide batch_size / 2 not batch_size？
1. Multiplying by constant doesn't really matter in terms of gradients.
2. During gradient calculation 2* will be got from x^2, i.e. /2 could be used to save extra calculation.

**Comment by anatolix** [comment]
> batches_x = np.empty(self.image_shape)

I use such things to prevent strange behaviour in multithreaded environment, actually keras iterator could be used in multithreaded mode, and there will be race condition if one thread will be copying data to gpu and other will be changing it same time
